test/crd.go

```diff
@@ -84,38 +84,13 @@ func getHelloWorldValidationPod(namespace, volumeClaimName string) *corev1.Pod {
 }
 
 func getHelloWorldTask(namespace string, args []string) *v1alpha1.Task {
-	return &v1alpha1.Task{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      hwTaskName,
-		},
-		Spec: v1alpha1.TaskSpec{
-			Steps: []corev1.Container{{
-				Name:    hwContainerName,
-				Image:   "busybox",
-				Command: args,
-			}},
-		},
-	}
+	return Task(hwTaskName, namespace,
+		TaskSpec(Step(hwContainerName, "busybox", Command(args...))),
+	)
```

Comment:
I actually like the definition of the task as a struct because it teaches new users how to create a task, or any of the other objects

Reply:
@pivotal-nader-ziada so I tend to agree on some level on that statement. That said, I think most of the new users will most likely look at samples (yaml) to know how to create objects, and for the rest, they may have the appropriate level to understand the builder code.

I'm not saying we should absolutely go for builder, if most of the contributor feel like we're loosing some value by introducing them (or not adding any value), I'll remove them :angel: :wink:

Reply:
(ps: maybe it makes more sense to have those builder in unit tests than here though :angel:)

Reply:
I like the idea of the builders in the unit tests, also not against them here as well, was just pointing out a benefit of showing the structure in a more clear way